core/services/feeds/service.go

```diff
@@ -274,10 +276,15 @@ func (s *service) SyncNodeInfo(ctx context.Context, id int64) error {
 
 		cfgMsgs = append(cfgMsgs, cfgMsg)
 	}
+	workflowKey, err := s.workflowKeyStore.Get(fmt.Sprintf("%d", id))
+	if err != nil {
+		return errors.Wrap(err, "could not fetch workflow key")
+	}
 
 	if _, err = fmsClient.UpdateNode(ctx, &pb.UpdateNodeRequest{
 		Version:      s.version,
 		ChainConfigs: cfgMsgs,
+		WorkflowKey:  workflowKey.PublicKeyString(),
```

Comment:
Why not just pass through the public key when you instantiate the service? This should be fairly easy to do and will have much less coupling

Reply:
you mean this won't change after instantiated? because this is called when we instantiate the service and whenever there is a change in the node config. If we want to use other method than `UpdateNode` this would involve a change on `chainlink-protos` to rollback adding this paramenter and adding it on other method. 

thinking out loud: don't we have an advantage if eg we want to rotate keys? I imagine we would somehow call this sync method instead of having to modify and re-start to register the new key